### PR TITLE
Use serverUID for web scp target

### LINKED ISSUE #22507: SSH Nodes: Download Files Ambiguous Error
Expected behavior:
When accessing a specific node in the Teleport UI, which shares a name with other nodes, it should allow you to download a file using the file transfer dialog without any issues.
Screenshot 2023-03-01 at 2 00 37 PM

Current behavior:
When starting a session through the Teleport UI to a specific node, which shares a name with other nodes, it fails to download any file using the file transfer dialog with the following error: 
Screenshot 2023-03-01 at 2 00 25 PM

Bug details:
- Teleport version: 12.0.x (tested on v12.0.1 and v12.0.4)
- Recreation steps: 
1. Add 2 Ubuntu 20.04 to your Teleport Cluster with the same node name
2. Attempt to download a file from any of the nodes 
- Debug logs: `failed connecting to node ninja-node4. err-node-is-ambiguous`

### PR DESCRIPTION
**Pull request by avatus**
Fixes https://github.com/gravitational/teleport/issues/22507

To fix this, I pass the unique server id present in the http request rather than a node name. A similar fix for Connect exists [here](https://github.com/gravitational/teleport/blob/master/lib/teleterm/clusters/cluster_file_transfer.go#L52)

### CODE REVIEW

**Comment by avatus** [comment]
Changing the `ExecuteSCP` signature wasn't my first option but then I realized it's only referenced in these two places (`files.upload` and `files.download`) so it was the simplest approach

File: lib/web/files.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ func (f *fileTransfer) download(req fileTransferRequest, httpReq *http.Request,
 		return trace.Wrap(err)
 	}
 
-	err = tc.ExecuteSCP(httpReq.Context(), cmd)
+	err = tc.ExecuteSCP(httpReq.Context(), req.server, cmd)
```
Comment (kimlisa) [inline lib/web/files.go:109]: if not too much trouble, can we rename `server` to `serverID`?

**Review by kimlisa — APPROVED** [review]

File: lib/web/files.go (hunk 1 above)
Comment (avatus) [reply lib/web/files.go]: No trouble at all! Fixed. much more clear this way.

File: lib/client/api.go [hunk 2]
```diff
@@ -1869,19 +1869,10 @@ func (tc *TeleportClient) ExecuteSCP(ctx context.Context, cmd scp.Command) (err
...
-	}
-	if len(nodeAddrs) == 0 {
-		return trace.BadParameter("no target host specified")
-	}
-
 	nodeClient, err := tc.ConnectToNode(
 		ctx,
 		proxyClient,
-		NodeDetails{Addr: nodeAddrs[0], Namespace: tc.Namespace, Cluster: tc.SiteName},
+		NodeDetails{Addr: serverAddr + ":0", Namespace: tc.Namespace, Cluster: tc.SiteName},
```
Comment (zmb3) [inline lib/client/api.go:1875]: Yeah I'm not sure what the `:0` is or why it's necessary. Once you figure it out, a comment is probably warranted.

Comment (avatus) [reply lib/client/api.go:1876]: Added a comment for the "why" but not really the "WHY why"... just the why _I'm_ doing it. Will still dig into it

File: lib/client/api.go [hunk 3]
```diff
@@ -1850,7 +1850,7 @@ func PlayFile(ctx context.Context, tarFile io.Reader, sid string) error {
 
 // ExecuteSCP executes SCP command. It executes scp.Command using
 // lower-level API integrations that mimic SCP CLI command behavior
-func (tc *TeleportClient) ExecuteSCP(ctx context.Context, cmd scp.Command) (err error) {
+func (tc *TeleportClient) ExecuteSCP(ctx context.Context, serverAddr string, cmd scp.Command) (err error) {
```
Comment (zmb3) [inline lib/client/api.go:1853]:
```suggestion
func (tc *TeleportClient) ExecuteSCP(ctx context.Context, serverAddr string, cmd scp.Command) error {
```

While you're here.

**Review by zmb3 — APPROVED** [review]

**Pull request merged**